### ISSUE TITLE
Xperia X does not wake for SMS when WiFi and mobile data are off

### ISSUE THREAD

**Issue by UniversalSuperBox** [ubports/ubuntu-touch#1562]
- Device: Xperia X
- Channel: armhf/devel
- Build: 2020-09-16

### Steps to reproduce

1. Turn off WiFi
1. Turn off mobile data
1. Unplug the device from power
1. Put the device to sleep with the power button, or allow it to fall asleep on a screen timeout
1. Send the device an SMS message from another place

### Expected behavior

The device wakes up and displays the SMS message.

### Actual behavior

The device may wake up and display the SMS message, or it may exhibit the following erroneous behavior:

* Turning on the display immediately shows the SMS message
* The screen does not light

### Logfiles and additional information

This issue was reported in https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-691728879 and I did some work here to reproduce. Unlike that reporter, I am unable to reproduce the issue if mobile data is enabled.

During testing, I found some interesting data about this issue. In particular, it occurs with the latest version of repowerd (`2018.04+ubports+0~20200812184122.5~1.gbp2a0e48`) and a build from January this year (`2018.01+ubports+0~20200122182344.11~1.gbpec4223`) that went unreleased in Ubuntu Touch images due to its lower version number. However, it does not occur with the `2018.03+1ubports1+0~20190409235621.9-1.gbp2f36d3` build from April 2019 and released with OTA-12. In other words, this likely broke when we released repowerd properly during the OTA-13 cycle.

This issue seems separate from https://github.com/ubports/ubuntu-touch/issues/1556 itself. Looking at the repowerd logs, it correctly holds the device awake until indicator-telephony-service dismisses its system state with `dbus_clearSysState` about 1.5 seconds later (host and PID information removed from log for brevity):

```
2020-09-16T08:28:27.167219-05:00: UnityScreenService: dbus_requestSysState(:1.64,telephony-service-indicator,1)
2020-09-16T08:28:27.169482-05:00: DefaultStateMachine[c1]: disallow_suspend
2020-09-16T08:28:27.169515-05:00: UnityScreenService: dbus_requestSysState(:1.64,telephony-service-indicator,1) => 4
2020-09-16T08:28:27.169519-05:00: LibsuspendSystemPowerControl: disallow_suspend(DefaultStateMachine)
2020-09-16T08:28:27.169537-05:00: LibsuspendSystemPowerControl: exiting suspend
2020-09-16T08:28:28.758610-05:00: UnityScreenService: dbus_clearSysState(:1.64,4)
2020-09-16T08:28:28.761846-05:00: DefaultStateMachine[c1]: allow_suspend
2020-09-16T08:28:28.761879-05:00: LibsuspendSystemPowerControl: allow_suspend(DefaultStateMachine)
2020-09-16T08:28:28.761896-05:00: LibsuspendSystemPowerControl: Preparing for suspend
```

I believe that what's happening here is the actual receipt of the notification is racing with the device going back to suspend. On every device I've tested *but* the Xperia X, Lomiri wins this race and is able to re-wakelock the device and turn on the screen. For whatever reason, it loses on the Xperia.

In this case, and certainly unlike #1556 where the user could go hours without seeing the message, the device *will* show the notification within 5 minutes of the message's receipt. The ubuntu-push service schedules wakeups every five minutes to check connectivity status and receive notifications. The notification is shown on that wakeup.

**Comment by Fuseteam** [comment]
device BQ E5
channel: armhf/devel
build: 2020-07-22
with wifi on it seems to take minute to wake from sleep and notify
without wifi it doesn't seem to wake it

**Comment by UniversalSuperBox** [comment]
Hi @Fuseteam, thanks for your report.

The July 22 image is affected by #1556, so you are seeing that issue on that build. This issue is separate to #1556, could you try to update and try the test again?

**Comment by Fuseteam** [comment]
will do, will try again once i'm up to date

**Comment by Fuseteam** [comment]
retested on build 2020-09-14 works as expected

**Comment by c4pp4** [comment]
The issue has no attention? No "critical" label? I think the device ignoring some incoming calls is useless. How can I help to fix the issue?

**Comment by Fuseteam** [comment]
@c4pp4 it definitely has attention, it marked as a bug. But there is so much going on right now. all the information needed to help fix the issue should be in the first post. if you can make sense out of it go ahead. if not feel free to ping the device maintainer if there's a way you can help them to solve the bug

**Comment by c4pp4** [comment]
> also this bug is not about the xperia x ignoring incoming calls it's about the screen not lighting up whenever it receives a sms; it silently receives sms's

@Fuseteam Look at the report link from "Logfiles and additional information" section of the first post. Or directly here https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-691728879 and here https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-693120572.

**Comment by Fuseteam** [comment]
@c4pp4 please look at the issue title
additionally this is the behavior that was reported in this issue as quoted
> 
    Actual behavior

    The device may wake up and display the SMS message, or it may exhibit the following erroneous behavior:

           Turning on the display immediately shows the SMS message
           The screen does not light
this is not issue 1556 it is issue 1562 that section only supposes it may be related to the other issue

**Comment by c4pp4** [comment]
@Fuseteam 
I offered help. The first post from UniversalSuperBox is about his experiences and that's why he is linking original experiences. I'm using the device as a daily driver and I'm sure it's the same problem. I missed many calls and I think it should be resolved as soon as possible otherwise the device is just a paperweight. Please, don't reply if you don't want to help, don't own the device, don't work on repowerd or don't have an idea how to resolve it. Thank you.

**Comment by UniversalSuperBox** [comment]
@c4pp4, I can corroborate that the Xperia X fails to wake for calls, but it fails to do so even when connected to Wi-Fi or cellular data. For that reason I believe it is a separate problem. I usually blame my cellular signal (3G here is awful), so I have never filed a bug. If you made a new report I'd be happy to confirm and add logs.

**Comment by UniversalSuperBox** [comment]
Also, please be kind when replying to issue reports. Most people here don't have English as their first language and are genuinely trying to help. Those of us who are paid already put in 8-12 hour days to bring you the OS. Volunteers just round up their workdays to 12 hours with us. We're trying our hardest to bring you a mobile platform, we're releasing it for free, so please keep that in mind while interacting here.

**Comment by c4pp4** [comment]
> @c4pp4, I can corroborate that the Xperia X fails to wake for calls, but it fails to do so even when connected to Wi-Fi or cellular data. For that reason I believe it is a separate problem.

https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-693120572:
> Cellular data - On
> Turn the device Off and On, don't lock/unlock it, just let it sleep via Lock when idle after 1 minute.
After the screen turns off, send a message to the device.
No notification shows up, you have to turn display on via button and the notification appears.

SMS notification fails even when connected to cellular data too. So I'm sure it's the same issue. Funny that you know about incoming call problem from my report and blame 3G signal.

>Also, please be kind when replying to issue reports.

I don't understand what is the problem?
BTW I'm a volunteer too, e.g. I spend many hours to resolve https://github.com/fredldotme/android_hardware_qcom_audio_aosp/commit/0bc085a5fb6cf848a53fcced544130e96e5fa1d7. If I use your words: I tried my hardest to bring you correct sample rate and I did it for free.

Maybe it was not clear so I can ask again:
### How can I help to fix the issue?

**Comment by Fuseteam** [comment]
hey i was trying to quantify the problem so that we can narrow down how this issue may be resolved. but if you take an attitude like that it's hard to get to the needed details.

remember to resolve an issue we need to find the cause. The first thing the devs do is to try to reproduce the bug. one they can reproduce it and monitor the logs to see what the cause is, so that can be debugged. if the experiences described above doesn't match yours exactly it might not actually be the same bug, which is why he asked you to open a new bug report. To know for sure if is indeed the same bug we'll need at least the logs.

so in short what you can do to help resolve the issue is provide some logs, so we can reproduce and debug. what is involved in debugging depends on the logs. but afaict at this point it seems like what dalton and you experience are two very similar but different bugs.

in either case based on dalton's original comment the repowerd logs will be insightful. those should be somewhere in `.cache/upstart`

**Comment by c4pp4** [comment]
@Fuseteam 
> so we can reproduce and debug

You are reproducing and debugging? You don't even read my post https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-693120572 - my log is already posted (/var/log/repowerd.log):
```
2020-09-16T04:31:01.513865+02:00: UnityScreenService: dbus_requestSysState(:1.70,telephony-service-indicator,1)
2020-09-16T04:31:01.515303+02:00: UnityScreenService: dbus_requestSysState(:1.70,telephony-service-indicator,1) => 1
2020-09-16T04:31:01.515406+02:00: DefaultStateMachine[c1]: disallow_suspend
2020-09-16T04:31:01.515434+02:00: LibsuspendSystemPowerControl: disallow_suspend(DefaultStateMachine)
2020-09-16T04:31:01.515451+02:00: LibsuspendSystemPowerControl: exiting suspend
2020-09-16T04:31:03.073656+02:00: UnityScreenService: dbus_clearSysState(:1.70,1)
2020-09-16T04:31:03.076848+02:00: DefaultStateMachine[c1]: allow_suspend
2020-09-16T04:31:03.076882+02:00: LibsuspendSystemPowerControl: allow_suspend(DefaultStateMachine)
2020-09-16T04:31:03.076899+02:00: LibsuspendSystemPowerControl: Preparing for suspend
```

**Comment by Fuseteam** [comment]
@c4pp4 there are more ways i can help no? and i have read your comment but i thought that log was for when the device doesn't wake for SMS, is there no difference at all in the logs when it misses calls?

**Comment by c4pp4** [comment]
@Fuseteam I understand what you mean, I'm trying to help as well, but tell me, I reported a bug - no notification, no incoming calls... I received a message it's filed as a new issue (see https://github.com/ubports/ubuntu-touch/issues/1556#issuecomment-693553769) and now I've found out nobody cares about incoming calls, just blaming 3G signal instead or what and I'm asking - is it professional behavior? Sorry if you feel offended by me I didn't mean it that way.

Data On, Wi-Fi Off

System freshly started and idled - No incoming call notification at all (60 seconds), only LED is blinking after that:
https://paste.ubuntu.com/p/ZSDtyxcRcf/

System locked via power button - No incoming call notification cca 45 seconds, after that incoming call is ringing:
https://paste.ubuntu.com/p/sRXNMjDvrn/

**Comment by Fuseteam** [comment]
@c4pp4 like i said the devs probably missed that part of your comment, which is why both dalton and i suggested opening a new issue. From what we can tell as it appears to be a separate issue from the reported/found issue, thanks for providing additional logs

**Comment by Fuseteam** [comment]
a second skim seems to show little difference between the two when it comes to the ofono service
```
2020-10-22T03:12:18.943746+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,true)
2020-10-22T03:12:18.944041+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,true)
2020-10-22T03:13:09.375050+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallAdded(/ril_0/voicecall01,incoming)
2020-10-22T03:13:54.463295+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,false)
2020-10-22T03:13:54.464431+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,false)
2020-10-22T03:14:10.036433+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallStateChanged(/ril_0/voicecall01,disconnected)
2020-10-22T03:14:10.038333+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallRemoved(/ril_0/voicecall01)
2020-10-22T03:14:15.292946+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,true)
2020-10-22T03:14:15.293249+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,true)
2020-10-22T03:14:19.345878+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,false)
2020-10-22T03:14:19.346199+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,false)
```

```
2020-10-22T02:50:15.636323+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,true)
2020-10-22T02:50:15.636665+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,true)
2020-10-22T02:50:59.429375+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallAdded(/ril_0/voicecall01,incoming)
2020-10-22T02:52:00.080915+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallStateChanged(/ril_0/voicecall01,disconnected)
2020-10-22T02:52:00.081314+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: dbus_CallRemoved(/ril_0/voicecall01)
2020-10-22T02:52:21.656767+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,false)
2020-10-22T02:52:21.657209+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,false)
2020-10-22T02:52:21.842819+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,true)
2020-10-22T02:52:21.843132+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,true)
2020-10-22T02:52:22.873191+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_0,false)
2020-10-22T02:52:22.873424+02:00 ubuntu-phablet repowerd[1505]: OfonoVoiceCallService
: set_fast_dormancy(/ril_1,false)
```

**Comment by c4pp4** [comment]
Wi-Fi On

System freshly started and idled - Incoming call notification is delayed by cca 4 seconds
https://paste.ubuntu.com/p/VZcfJNfRTF/

**Comment by Fuseteam** [comment]
that sounds like its working more or less. the question is why does wifi on matter.
do you get a the same results with wifi off data off and you do with wifi off data on?

perhaps we can narrow it down further than way

**Comment by fredldotme** [comment]
IIRC enabling WIFI aggressively keeps the device awake in bcmdhd every few seconds for every signal transmission, whereas the modem driver doesn't seem to be waking as aggressively. That's the first thing I would compare, in the kernel, so whether it's due to wake locks or wake_up calls or whatnot.

**Comment by c4pp4** [comment]
>### Logfiles and additional information
> However, it does not occur with the `2018.03+1ubports1+0~20190409235621.9-1.gbp2f36d3` build from April 2019 and released with OTA-12. In other words, this likely broke when we released repowerd properly during the OTA-13 cycle.

I was trying to use 2018.03+1ubports1+0~20190409235621.9-1.gbp2f36d3 version. It behaves the same way as the version from OTA-13 release - system freshly started and idled - no incoming call notification, sms notification delayed.

**Comment by c4pp4** [comment]
- arm64/devel
- build 2020-10-31
- data On, Wi-Fi Off

System **freshly** started and idled or locked via power button - the same as previous - no incoming call notification at all, only LED is blinking after that.

After the first failed incoming call notification **there is no problem** with the next ones.

https://paste.ubuntu.com/p/pW7TWRmHFM/

**Comment by boriseto** [comment]
Using the Xperia X without turning off data or wifi at any point. I only thought that I missed the calls because of not listening to the phone ring. Can you please at least confirm that the calls are shown in recent calls as missed? Are there calls that aren't shown at all there too?

**Comment by c4pp4** [comment]
@boriseto
> Can you please at least confirm that the calls are shown in recent calls as missed?

 I can confirm it.

**Comment by c4pp4** [comment]
Another example:

Set alarm, restart system and lock it via power button - alarm wakes the device up without problem (screen, sound and vibrations), after that lock it again and make incoming call - no notification at all, only LED is blinking after that.
21:15:01 alarm
21:15:38 incoming call
https://paste.ubuntu.com/p/QwSmNTcpy4/

**Comment by UniversalSuperBox** [comment]
I believe this should be fixed by https://github.com/Halium/halium-devices/commit/e12234ea59f4282b09f6251927efd942f6c5eafc. The new kernel enabled by this change has been shipped on the devel channel today.

I've tested with Wi-Fi and mobile data off, I now receive calls and texts immediately with a notification buzz or ringtone. Please give it a try and let me know if it works for you as well!

Since it's working for me, I'll close this report. If it does not work for you, I'll reopen it and we can continue investigating.

**Comment by c4pp4** [comment]
@UniversalSuperBox: I can confirm It's resolved.

Thank you very much @fredldotme.
